tests/unit/test_transform.py

```diff
@@ -0,0 +1,64 @@
+import unittest
+from pathlib import PosixPath
+
+from ops.charm import CharmBase
+from ops.testing import Harness
+
+from lib.charms.prometheus_k8s.v0.prometheus_scrape import (
+    MissingTransformResourceError,
+    _TransformationProvider,
+)
+
+META = """
+resources:
+  promql-transform-amd64:
+    type: file
+    description: test
+"""
+
+
+# noqa: E302
+# pylint: disable=too-few-public-methods
+class TransformProviderCharm(CharmBase):
+    """Container charm for running the integration test."""
+
+    def __init__(self, *args):
+        super().__init__(*args)
+        self.transformer = _TransformationProvider(self)
+
+
+class TestTransform(unittest.TestCase):
+    """Test that the promql-transform implementation works."""
+
+    def setUp(self):
+        self.harness = Harness(TransformProviderCharm, meta=META)
+        self.harness.set_model_name("transform")
+        self.addCleanup(self.harness.cleanup)
+        self.harness.add_resource("promql-transform-amd64", "dummy resource")
+        self.harness.begin()
+
+    # pylint: disable=protected-access
```

Comment:
`disable protected access` could probably be applied to everything under `tests/`. [Example](https://github.com/canonical/alertmanager-operator/blob/bf95b19feb05b21f945559950f2bc47132747648/pyproject.toml#L30)